src/App.tsx

```diff
@@ -15,71 +15,63 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const [selectedGood, chosenGood] = useState(goods[8]);
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const clear = () => {
+    chosenGood('');
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      <h1 className="title is-flex is-align-items-center">
+        { selectedGood === ''
+          ? 'No goods selected'
+          : `${selectedGood} is selected`}
+        {selectedGood !== '' && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            aria-label="button"
+            onClick={clear}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+      <table className="table">
+        <tbody>
+          {goods.map((item) => {
+            return (
+              <tr
+                data-cy="Good"
+                className={` ${selectedGood === item ? 'has-background-success-light' : ''}`}
+                key={item}
+              >
+                <td>
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className={`button  ${selectedGood === item ? 'is-info' : ''}`}
+                    onClick={() => {
+                      chosenGood(item);
+                    }}
+                  >
+                    {selectedGood === item ? '-' : '+'}
```

Comment:
Move selectedGood === item to a variable. You use it not once.

Reply:
not fixed
<img width="797" alt="image" src="https://user-images.githubusercontent.com/53611910/232252520-b48623bb-f85c-48cc-a61e-4afadc722cab.png">